src/Sylius/Bundle/ApiBundle/CommandHandler/RegisterShopUserHandler.php

```diff
@@ -15,42 +15,30 @@
 
 use Doctrine\Persistence\ObjectManager;
 use Sylius\Bundle\ApiBundle\Command\RegisterShopUser;
-use Sylius\Component\Core\Model\CustomerInterface;
+use Sylius\Bundle\ApiBundle\Provider\CustomerProviderInterface;
 use Sylius\Component\Core\Model\ShopUserInterface;
-use Sylius\Component\Core\Repository\CustomerRepositoryInterface;
 use Sylius\Component\Resource\Factory\FactoryInterface;
-use Sylius\Component\User\Canonicalizer\CanonicalizerInterface;
 use Symfony\Component\Messenger\Handler\MessageHandlerInterface;
 
-class RegisterShopUserHandler implements MessageHandlerInterface
+final class RegisterShopUserHandler implements MessageHandlerInterface
 {
-    /** @var CanonicalizerInterface */
-    protected $canonicalizer;
-
     /** @var FactoryInterface */
     protected $shopUserFactory;
 
-    /** @var FactoryInterface */
-    protected $customerFactory;
-
-    /** @var CustomerRepositoryInterface */
-    protected $customerRepository;
-
     /** @var ObjectManager */
     protected $shopUserManager;
 
+    /** @var CustomerProviderInterface */
+    protected $customerProvider;
```

Comment:
```suggestion
    private $customerProvider;
```
this should be applied to all properties